resources/js/components/ui/form/Checkbox.tsx

```diff
@@ -0,0 +1,58 @@
+import {Component, JSX, createEffect, splitProps} from "solid-js";
+import * as checkbox from "@zag-js/checkbox";
+import {normalizeProps, useMachine} from "@zag-js/solid";
+import {createMemo, createUniqueId} from "solid-js";
+import s from "./Checkbox.module.scss";
+import {ValidationMessages} from "./ValidationMessages";
+import {useFieldLabel} from "./FieldLabel";
+
+export interface CheckboxProps extends Omit<JSX.InputHTMLAttributes<HTMLInputElement>, "onChange"> {
+  name: string;
+  label?: string;
+  disabled?: boolean;
+  onChange?: checkbox.Context["onChange"];
+  indeterminate?: boolean;
+}
+
+/**
+ * Wrapper of Zag's Checkbox
+ *
+ * Intended for use with FelteForm (handles validation messages)
+ */
+export const Checkbox: Component<CheckboxProps> = (props) => {
+  const [machineProps, inputProps] = splitProps(props, ["onChange"]);
+
+  const [state, send] = useMachine(
+    checkbox.machine({
+      id: createUniqueId(),
+      onChange: (details) => machineProps.onChange?.(details),
+    }),
+    {
+      context: {
+        name: props.name,
+      },
+    },
+  );
+
+  const api = createMemo(() => checkbox.connect(state, send, normalizeProps));
+
+  createEffect(() => {
+    if (inputProps.checked === undefined) return;
+    api().setChecked(inputProps.checked);
```

Comment:
Tak chyba nie możesz zrobić. Trzeba pozwolić felte zarządzać tym stanem a nie jakimś propsom.

Ja to bym w ogóle sugerował zrobić to przez createField, wtedy się nie martwisz jakimiś typami eventów i czymś, tylko po prostu implementujesz ten adapter. Chociaż fakt że piszą że jeśli na dole jest natywna kontrolka to powinno to działać z felte.